opentok-whiteboard.js

```diff
@@ -226,31 +246,56 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
                 }
             });
 
+            /*
+             *    The Nuts
+             *    During the process of drawing, we collect coordinates on every [mouse|touch]move event.
+             *    These events occur as fast as the browser can create them, and is computer/browser dependent
+             *
+             */
+
             angular.element(canvas).on('mousedown mousemove mouseup mouseout touchstart touchmove touchend touchcancel',
+
               function (event) {
-                if ((event.type === 'mousemove'|| event.type === 'touchmove') && !client.dragging) {
+                if ((event.type === 'mousemove' || event.type === 'touchmove' || event.type === 'mouseout') && !client.dragging) {
                     // Ignore mouse move Events if we're not dragging
                     return;
                 }
+
                 event.preventDefault();
-                
+
                 var offset = angular.element(canvas).offset(),
                     scaleX = canvas.width / element.width(),
-                    scaleY = canvas.height / element.height();
-                var offsetX, offsetY, x, y;
-                
-                switch (event.type) {
-                case 'mousedown':
-                case 'touchstart':
+                    scaleY = canvas.height / element.height(),
                     offsetX = event.offsetX || event.originalEvent.pageX - offset.left ||
-                        event.originalEvent.touches[0].pageX - offset.left,
+                       event.originalEvent.touches[0].pageX - offset.left,
                     offsetY = event.offsetY || event.originalEvent.pageY - offset.top ||
-                        event.originalEvent.touches[0].pageY - offset.top,
+                       event.originalEvent.touches[0].pageY - offset.top,
                     x = offsetX * scaleX,
-                    y = offsetY * scaleY;
+                    y = offsetY * scaleY,
+                    mode = scope.erasing ? 'eraser' : 'pen',
+                    update;
+
+                switch (event.type) {
+                case 'mousedown':
+                case 'touchstart':
+                    // Start dragging
                     client.dragging = true;
+
                     client.lastX = x;
                     client.lastY = y;
+
+                    update = {
+                        id: OTSession.session && OTSession.session.connection &&
```

Comment:
I originally started with that, but since the update is only a part of the whole path and paper.js assigns new paths a unique id already it I think worked out better to use that id.